packages/template-retail-react-app/scripts/extract-default-messages.js

```diff
@@ -0,0 +1,92 @@
+ * If a file is overridden, it won't extract messages from that file in the base template
+ */
+const {exec} = require('child_process')
+const fs = require('fs')
+const path = require('path')
+const packagePath = path.join(process.cwd(), 'package.json')
+const pkgJSON = JSON.parse(fs.readFileSync(packagePath))
+
+const getAllFilesByExtensions = (dirPath, arrayOfFiles = [], extensions = []) => {
+    const files = fs.readdirSync(dirPath)
+
+    files.forEach(function (file) {
+        if (fs.statSync(path.join(dirPath, file)).isDirectory()) {
+            arrayOfFiles = getAllFilesByExtensions(
+                path.join(dirPath, file),
+                arrayOfFiles,
+                extensions
+            )
+        } else {
+            arrayOfFiles.push(path.join(dirPath, file))
+        }
+    })
+    if (extensions.length) {
+        return arrayOfFiles.filter((filePath) => {
+            const getExtension = path.extname(filePath).replace('.', '')
+            return extensions.includes(getExtension)
+        })
+    }
+
+    return arrayOfFiles
+}
+
+try {
+    const overridesDir = pkgJSON.ccExtensibility?.overridesDir
+    if (!overridesDir) {
+        const command =
+            "formatjs extract 'app/**/*.{js,jsx}' --out-file translations/en-US.json --id-interpolation-pattern [sha512:contenthash:base64:6]"
+        exec(command, (err) => {
+            if (err) {
+                console.error(err)
+            }
+        })
+    } else {
+        const overridesPath = path.join(process.cwd(), pkgJSON.ccExtensibility?.overridesDir)
+        // get all the files in extended app
+        const files = getAllFilesByExtensions(
+            path.join(overridesPath, 'app'),
+            [],
+            ['js', 'jsx', 'ts', 'tsx']
+        )
+        // get the file names that are overridden in base template
+        const overriddenFiles = files
+            .map((path) => {
+                const replacedPath = path.replace(
+                    overridesDir,
+                    `node_modules/${pkgJSON.ccExtensibility?.extends}`
+                )
+                // check if this file does exist in base template
+                const isFileExist = fs.existsSync(replacedPath)
+                return isFileExist ? replacedPath : ''
+            })
+            .filter(Boolean)
+        // rename the files needs to ignore to have .ignore extensions
+        overriddenFiles.forEach((filePath) => {
+            fs.rename(filePath, `${filePath}.ignore`, (err) => err && console.error(err))
+        })
+
+        const extractCommand = `formatjs extract "./node_modules/${pkgJSON.ccExtensibility?.extends}/app/**/*.{js,jsx}" "${pkgJSON.ccExtensibility?.overridesDir}/app/**/*.{js,jsx}" --ignore "./node_modules/${pkgJSON.ccExtensibility?.extends}/app/**/*.ignore" --out-file translations/en-US.json --id-interpolation-pattern [sha512:contenthash:base64:6]`
```

Comment:
Since we already renamed the file extension to `js.ignore`, is it still being included in the globe `/app/**/*.{js,jsx}`? I'm just wondering whether we still need the `--ignore` or not.

Reply:
Good point, it seems like changing the file extension works, we don't need to use --ignore